### PR TITLE
Migrate scripts from using os.path to using Pathlib

### PR DESCRIPTION
**Pull request by tim-fry**
This pull request fix openmc-dev/data#17:

I've removed use of os.path from the following 5x convert/generate scripts:
 - convert_fendl.py
 - convert_jeff32.py
 - convert_tendl.py
 - generate_cendl.py
 - generate_jendl.py

This includes replacing the use of the glob module for file searching with the Path.glob function.

Testing was performed by running the original and modified scripts and performing the following checks:

 - That the console logs produced by both scripts were identical
 - That the files extracted before processing were identical
 - That the number and size of hdf5 files created were identical
 - That the cross_sections.xml files created were identical
 - That a subset of created hdf5 files created contained identical data. This is done using the h5diff utility because hdf5 files report differences using diff.

Where the generate_* scripts take considerable time running njoy to process the downloaded data, these scripts were testing using specially created subsets of the downloaded data designed to test the different parts of the script. For the generate_cendl.py script this included the files which require file specific fixes.

### CODE REVIEW

File: convert_jeff32.py [hunk 1]
```diff
@@ -173,18 +171,17 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
 # Sort temperatures from lowest to highest
 for name, filenames in sorted(tables.items()):
     filenames.sort(key=lambda x: int(
-        os.path.split(x)[1].split('-')[1].split('.')[0]))
+        x.name.split('-')[1].split('.')[0]))
 
 for name, filenames in sorted(tables.items()):
     # Convert first temperature for the table
-    print('Converting: ' + filenames[0])
+    print('Converting: ' + str(filenames[0]))
```
Comment (paulromano) [inline convert_jeff32.py]:
```suggestion
    print(f'Converting: {filenames[0]}')
```

File: convert_jeff32.py [hunk 2]
```diff
@@ -194,7 +191,7 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
 
     # For each higher temperature, add cross sections to the existing table
     for filename in filenames[1:]:
-        print('Adding: ' + filename)
+        print('Adding: ' + str(filename))
```
Comment (paulromano) [inline convert_jeff32.py]:
```suggestion
        print(f'Adding: {filename}')
```

File: convert_tendl.py [hunk 3]
```diff
@@ -118,35 +115,33 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
-    if library_name == 'tendl' and args.release == '2017' and os.path.basename(filename) == 'B010':
+    if library_name == 'tendl' and args.release == '2017' and filename.name == 'B010':
         text = open(filename, 'r').read()
         if text[423:428] == '86843':
             print('Manual fix for incorrect value in ACE file') # see OpenMC user group issue for more details
             text = ''.join(text[:423])+'86896'+''.join(text[428:])
             open(filename, 'w').write(text)
 
-    print('Converting: ' + filename)
+    print('Converting: ' + str(filename))
```
Comment (paulromano) [inline convert_tendl.py]:
```suggestion
    print(f'Converting: {filename}')
```

File: generate_cendl.py [hunk 4]
```diff
@@ -89,47 +88,44 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
-    os.mkdir(args.destination)
+args.destination.mkdir(parents=True, exist_ok=True)
 
 library = openmc.data.DataLibrary()
 
 for filename in sorted(neutron_files):
 
     # this is a fix for the CENDL 3.1 release where the 22-Ti-047.C31 and 5-B-010.C31 files contain non-ASCII characters
-    if library_name == 'cendl' and args.release == '3.1' and os.path.basename(filename) in  ['22-Ti-047.C31','5-B-010.C31']:
+    if library_name == 'cendl' and args.release == '3.1' and filename.name in  ['22-Ti-047.C31','5-B-010.C31']:
```
Comment (paulromano) [inline generate_cendl.py]:
```suggestion
    if library_name == 'cendl' and args.release == '3.1' and filename.name in  ['22-Ti-047.C31', '5-B-010.C31']:
```

File: generate_cendl.py [hunk 5]
```diff
@@ -89,47 +88,44 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
-        if os.path.basename(filename) == '22-Ti-047.C31':
+        if filename.name == '22-Ti-047.C31':
             text[205] = ' 8) YUAN Junqian,WANG Yongchang,etc.               ,16,(1),57,92012228 1451  205'
-        if os.path.basename(filename) == '5-B-010.C31':
+        if filename.name == '5-B-010.C31':
             text[203] = '21)   Day R.B. and Walt M.  Phys.rev.117,1330 (1960)               525 1451  203'
         open(filename, 'w').write('\r\n'.join(text))
 
-    print('Converting: ' + filename)
+    print('Converting: ' + str(filename))
```
Comment (paulromano) [inline generate_cendl.py]:
```suggestion
    print(f'Converting: {filename}')
```

File: generate_jendl.py [hunk 6]
```diff
@@ -91,37 +90,34 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
-if not os.path.isdir(args.destination):
-    os.mkdir(args.destination)
+args.destination.mkdir(parents=True, exist_ok=True)
 
 library = openmc.data.DataLibrary()
 
 for filename in sorted(neutron_files):
 
-    print('Converting: ' + filename)
+    print('Converting: ' + str(filename))
```
Comment (paulromano) [inline generate_jendl.py]:
```suggestion
    print(f'Converting: {filename}')
```

**Comment by tim-fry** [comment]
Hi Paul, thanks for the pointers. I've committed those changes as requested.

**Comment by paulromano** [comment]
Thanks for the updates @tim-fry. It looks like there is a conflict in convert_tendl.py due to #16 being merged. Once that's resolved, I think you may need to look through that script again to weed out any instances of the other `os` based functions that might have gotten introduced.

**Comment by tim-fry** [comment]
No problem Paul (I wasn't sure from the message where the merge needed to occur). Leave it with me.

**Comment by Shimwell** [comment]
congrats @tim-fry this contribution is very much appreciated

**Pull request merged**